src/Options.js

```diff
@@ -43,12 +54,27 @@ const sizeAtStyle = function(size, style) {
  * different properties, call a `.having*` method.
  */
 class Options {
-    constructor(data) {
+    style: StyleInterface;
+    color: ?string;
```

Comment:
I this is changed to `color: string | void` then the `color` prop in `OptionsData` could be `style?: string` like the other optional properties.  This would also match existing code in https://github.com/Khan/KaTeX/blob/588f5a1ee64b1c4ebde18b1af139ad1e6d00ce63/src/ParseNode.js#L19-L21 for optional properties on classes.

Reply:
I've switched to `T | void` for nullable types, but this still doesn't allow `color?: string` in `OptionsData`.

In `Options.extend`, `data` must be compatible with the type `OptionsData` in order to be passed into `new Options(data)`, but if `this.color` is of type `?string`, you cannot assign it to property `color` of an `OptionsData`, which must have type `string`. The same applies for `font`.